moto/glue/utils.py

```diff
@@ -0,0 +1,378 @@
+                    raise InvalidStateException(
+                        "An error occurred (InvalidStateException) when calling the"
+                        f" GetPartitions operation: {e}"
+                    )
+
+        # also raised for unpartitioned tables
+        self._raise_unknown_column()
+
+    def leval(self, part_keys: List[Dict[str, str]], literal: Any) -> Any:
+        # evaluate literal by simulating partition input
+        for key in part_keys:
+            if self.ident == key["Name"]:
+                try:
+                    return _cast(key["Type"], literal)
+                except ValueError as e:
+                    # expression literal cannot be cast to current schema
+                    raise InvalidInputException(
+                        "An error occurred (InvalidInputException) when calling the"
+                        f" GetPartitions operation: {e}"
+                    )
+
+        # any unknown column is detected during eval already
+        raise NotImplementedError("leval called before eval")
+
+    def type_(self, part_keys: List[Dict[str, str]]) -> str:
+        for key in part_keys:
+            if self.ident == key["Name"]:
+                return key["Type"]
+
+        self._raise_unknown_column()
+
+    def _raise_unknown_column(self):
+        raise InvalidInputException(
+            "An error occurred (InvalidInputException) when calling the"
+            f" GetPartitions operation: Unknown column '{self.ident}'"
+        )
+
+
+class _IsNull(_Expr):
+    def __init__(self, tokens: ParseResults):
+        self.ident: _Ident = tokens[0]
+
+    def eval(self, part_keys: List[Dict[str, str]], part_input: Dict[str, Any]) -> bool:
+        return self.ident.eval(part_keys, part_input) is None
+
+
+class _IsNotNull(_IsNull):
+    def eval(self, part_keys: List[Dict[str, str]], part_input: Dict[str, Any]) -> bool:
+        return not super().eval(part_keys, part_input)
+
+
+class _BinOp(_Expr):
+    def __init__(self, tokens: ParseResults):
+        self.ident: _Ident = tokens[0]
+        self.bin_op: str = tokens[1]
+        self.literal: Any = tokens[2]
+
+    def eval(self, part_keys: List[Dict[str, str]], part_input: Dict[str, Any]) -> bool:
+        ident = self.ident.eval(part_keys, part_input)
+
+        # simulate partition input for the lateral
+        rhs = self.ident.leval(part_keys, self.literal)
+
+        return {
+            "<>": operator.ne,
+            ">=": operator.ge,
+            "<=": operator.le,
+            ">": operator.gt,
+            "<": operator.lt,
+            "=": operator.eq,
+        }[self.bin_op](ident, rhs)
+
+
+class _Like(_Expr):
+    def __init__(self, tokens: ParseResults):
+        self.ident: _Ident = tokens[0]
+        self.literal: str = tokens[2]
+
+    def eval(self, part_keys: List[Dict[str, str]], part_input: Dict[str, Any]) -> bool:
+        warnings.warn("LIKE expression conversion to regex is experimental")
```

Comment:
Same here w.r.t. the warning - this should be part of the documentation, but we don't have to warn people every time. Either it works correctly, or it doesn't.

Reply:
Agree